docs/apm-mutating-webhook.asciidoc

```diff
@@ -0,0 +1,230 @@
+all container images contained in the original incoming object.
+. Copy the agent binary from the init container image into the shared volume,
+making it available to the other container images.
+. Update the environment variables in the container images to configure
+auto-instrumentation with the copied agent binary
+
+TIP: To learn more about mutating webhooks,
+see the {kube-admin-docs}[Kubernetes Admission controller documentation].
+
+[[apm-helm-chart]]
+=== Helm chart
+
+The Helm chart manages the configuration of all associated manifest files for the
+webhook receiver, including generating certificates for securing communication
+between the Kubernetes API server and the webhook receiver.
+
+TIP: To learn more about Helm charts, see the {helm-docs}[Helm documentation].
+
+// Break content below to a new page
+
+[[apm-get-started-webhook]]
+== Get started with the webhook
+
+To instrument and configure your application pods, complete the following steps:
+
+. <<apm-webhook-clone-repo>>
+. <<apm-webhook-configure-helm>>
+. <<apm-webhook-install-helm>>
+. <<apm-webhook-add-pod-annotation>>
+// Anything next? Restart the deployment maybe?
+
+[float]
+[[apm-webhook-clone-repo]]
+=== Clone the repository
+
+Clone the `elastic/apm-mutating-webhook` repository:
+
+[source,bash]
+----
+git clone git@github.com:elastic/apm-mutating-webhook.git
+cd apm-mutating-webhook
+----
+
+[float]
+[[apm-webhook-configure-helm]]
+=== Configure the webhook with a Helm values file
+
+The webhook is installed with a Helm chart.
+You can provide a custom webhook configuration using a Helm values file.
+Elastic provides a https://github.com/elastic/apm-mutating-webhook/blob/main/custom.yaml[`custom.yaml`] file as a starting point.
+
+Here's an example of a `custom.yaml` file that instruments a pod with the Elastic APM Java agent:
+
+[source,yaml]
+----
+apm:
+  secret_token: SuP3RT0K3N <1>
+  namespaces: <2>
+    - default
+    - my-name-space-01
+    - my-name-space-02
+webhookConfig:
+  agents:
+    java:
+      image: docker.elastic.co/observability/apm-agent-java:latest
+      artifact: "/usr/agent/elastic-apm-agent.jar"
+      environment:
+        JAVA_TOOL_OPTIONS: "-javaagent:/elastic/apm/agent/elastic-apm-agent.jar"
+        ELASTIC_APM_SERVER_URL: "https://apm-example.com:8200" <3>
+        ELASTIC_APM_ENVIRONMENT: "prod"
+        ELASTIC_APM_LOG_LEVEL: "info"
+----
+<1> The `secret_token` for your deployment. Use `api_key` if using an API key instead.
+<2> If you're using a secret token or API key to secure your deployment, you must list
+all of the namespaces where you want to auto-instrument pods. The secret token or API key
+will be stored as Kubernetes Secrets in each namespace.
+<3> Elastic APM agent environment variables—for example, the APM Server URL, which specifies the URL and port of your APM integration or server.
+
+NOTE:: The `artifact` and `JAVA_TOOL_OPTIONS` keys should not be edited.
```

Comment:
We mention that the `artifact` and `JAVA_TOOL_OPTIONS` keys should not be edited. Are they required? The `JAVA_TOOL_OPTIONS` is Java specific, but what about `artifact`? Neither are in the [custom.yaml](https://github.com/elastic/apm-mutating-webhook/blob/main/custom.yaml) example file.

Reply:
Also, I'm guessing the `image` shouldn't be edited either. Is that not noted because it's assumed?